docs/proposals/moving-to-dynamodb.md

```diff
@@ -0,0 +1,358 @@
+--------------------- | --------------------- | ------------------ | ----------------
+key                   | **--**                | String **(pk)**    | `${pkg}!${env}!${version}`
+pkg                   | text **(pk)**         | String             |
+env                   | text **(pk)**         | String             |
+version               | text **(pk)**         | String             |
+locale                | text                  | String             |
+error                 | boolean               | Boolean            |
+message               | text                  | String             |
+details               | text                  | String             |
+create_date           | timestamp             | String             | `dynamodb`'s createdAt
+event_id              | unqiue_timestamp **(ck)** | String **(sk)**    | `dynamodb`'s timeUUID
+
+### StatusCounter (`status_counter`)
+
+Column                | Cassandra Type        | DynamoDB Type      | Notes
+--------------------- | --------------------- | ------------------ | ----------------
+key                   | **--**                | String **(pk)**    | `${pkg}!${env}!${version}`
+pkg                   | text **(pk)**         | String             |
+env                   | text **(pk)**         | String             |
+version               | text **(pk)**         | String             |
+count                 | counter               | Number             |
+
+## Affected modules
+
+The following `warehouse.ai` modules will be affected by this change:
+
+1. [`warehouse-models`]
+1. [`@wrhs/release-line`]
+1. [`bffs`]
+1. [`carpenterd-worker`]
+1. [`carpenterd`]
+1. [`feedsme`]
+1. [`warehouse.ai`]
+1. [`warehouse.ai-status-models`]
+1. [`warehouse.ai-status-api`]
+
+## Backwards compatibility
+
+There are no plans to provide this change in a backwards compatible way. Each
+of the affected modules will published after the change with a new semver-major
+version. All interactions with the data-store flow through `warehouse-models`
+or `warehouse.ai-status-models`, that should continue to be the case.  We will
+strive to keep the API surface mostly compatible, but some changes will be
+required (like not having a `package_cache` table, partition keys changing,
+etc.).
+
+## Data Modeling
+
+As part of this transition and the move to a more `serverless` style we will
+just directly use the [`dynamodb`] library from the models packages rather than
+trying to adapt `datastar` to work with dynamodb.
+
+## Future Serverless Considerations
+
+Out of scope for this proposal, but interesting potential future work related
+to serverless could include:
+
+1. Changing the messaging architecture
+   > Currently `warehouse.ai` uses a combination of direct http calls and NSQ
+   > for messaging between components, We should consider moving off that for
+   > something like [AWS Step Functions], [Amazon SNS], [Amazon SQS], etc.
+1. Builds on demand
```

Comment:
what about data migration for existing Cassandra databases? Do we consider that out of scope, should we write a script to accommodate for it?

Reply:
I tried to address that here.  We will have one, based on our custom scripts, but since our scripts are intermingled with other pieces and there's limited use outside us right now, there's probably not much utility in trying to rework it to be more open source friendly.  We can do that if we think there's utility in it. I'm all for that, its just more work that might not be really useful to anyone.